source/adapters/hip/usm.cpp

```diff
@@ -385,8 +385,6 @@ UR_APIEXPORT ur_result_t UR_APICALL urUSMPoolCreate(
                    ///< ::ur_usm_pool_limits_desc_t
     ur_usm_pool_handle_t *Pool ///< [out] pointer to USM memory pool
 ) {
-  // Without pool tracking we can't free pool allocations.
-#ifdef UMF_ENABLE_POOL_TRACKING
```

Comment:
Last time I checked, `UMF_ENABLE_POOL_TRACKING` was disabled in HIP (note the lack of CMake adding the compile definitions). Has something changed such that we're able to enabled pool tracking by default in HIP?

Reply:
I forget what the specific problem was, but IIRC only some versions of HIP were having problems. See #1479 and #1689. I couldn't make much progress with it just ended up prioritising something else, but I see in my notes that I put it down to HIP 5.7.1 crashing in CI (at the time) whereas locally 6.0+ was okay.

I don't know what we're testing in CI these days, I'm afraid.

Reply:
Also, we don't run HIP E2E testing for DPC++, so I wonder if that's something we ought to verify too. I am unaware what the UMF tracking problems with HIP were. I presume the passing CI, based on Fraser's previous observations, could be due to testing newer version of HIP these days.

Reply:
I've rebased that old PR of mine, just in case it is able to provide some more context/data about what's going on.

Reply:
Hey @lukaszstolarczuk 
This PR https://github.com/oneapi-src/unified-runtime/pull/1689 removes some more USMAlloc tests from the match files and they seem to fail with enabled memory pools (via `UMF_ENABLE_POOL_TRACKING`). Maybe they should remain as known failures for now and still have memory pools enabled by default but I am not sure of any implications.
I think I'll use yours and @frasercrmck's best judgement on this matter.

Reply:
I think, given that this PR is not NFC for HIP, we should see some changes to the match files. The PR description should also reflect this. If there are still failures in the USM tests with this PR (as I suspect there will be, given the problems I had on my PR) then we need to account for them somehow. This might be tricky given that CI isn't currently testing a supported GPU.